src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,42 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int ADULT_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkAge(user);
+        checkLogin(user);
+        checkPassword(user);
+        return storageDao.add(user);
+    }
+
+    private void checkAge(User user) {
+        if (user.getAge() < ADULT_AGE) {
+            throw new RuntimeException("Can't register user age under 18");
+        }
+        if (user.getAge() >= ADULT_AGE) {
+            storageDao.add(user);
+        }
+    }
+
+    private void checkLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new RuntimeException("Login can't be NULL");
+        }
+        if (user.equals(storageDao.get(user.getLogin()))) {
+            throw new RuntimeException("User with such login already exists");
+        }
+    }
+
+    private void checkPassword(User user) {
```

Comment:
You wrote a test that checks password is empty, but you haven't this check in  implementation, and you can the same check in checklogin method and write test about that.

Reply:
We already have a limit of password length which is 6. Empty string is under that limit? so I think there is no need to check it.  But in login we need this check, because there is no length requrement of login.

Reply:
Sorry, you're right